src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntTests.cs

```diff
@@ -177,24 +213,69 @@ private static void VerifyTypes(NamedTypeSymbol underlyingType, NamedTypeSymbol
             VerifyType(underlyingType, signed, isNativeInt: false);
             VerifyType(nativeIntegerType, signed, isNativeInt: true);
 
-            Assert.Empty(nativeIntegerType.MemberNames);
             Assert.Empty(nativeIntegerType.GetTypeMembers());
-            Assert.Empty(nativeIntegerType.GetMembers());
+
+            var memberNames = nativeIntegerType.MemberNames;
+            var members = nativeIntegerType.GetMembers();
+            Assert.Equal(members.SelectAsArray(m => m.Name), memberNames);
+
+            var baseMembers = members.SelectAsArray(m => ((MethodSymbol)m).OverriddenMethod);
+            Assert.Equal(
+                new[] { "int System.ValueType.GetHashCode()", "bool System.ValueType.Equals(object obj)", "string System.ValueType.ToString()" },
+                baseMembers.SelectAsArray(m => m.ToDisplayString(FormatWithSpecialTypes)));
 
             // PROTOTYPE: Include certain interfaces defined on the underlying underlyingType.
             Assert.Empty(nativeIntegerType.InterfacesNoUseSiteDiagnostics());
 
             Assert.Same(underlyingType, underlyingType.AsNativeInt(false));
             Assert.Same(nativeIntegerType, nativeIntegerType.AsNativeInt(true));
-            Assert.Equal(nativeIntegerType, underlyingType.AsNativeInt(true));
-            Assert.Equal(underlyingType, nativeIntegerType.AsNativeInt(false));
+            Assert.Same(nativeIntegerType, underlyingType.AsNativeInt(true));
+            Assert.Same(underlyingType, nativeIntegerType.AsNativeInt(false));
             VerifyEqualButDistinct(underlyingType, underlyingType.AsNativeInt(true));
             VerifyEqualButDistinct(nativeIntegerType, nativeIntegerType.AsNativeInt(false));
             VerifyEqualButDistinct(underlyingType, nativeIntegerType);
 
             VerifyTypes(underlyingType.GetPublicSymbol(), nativeIntegerType.GetPublicSymbol(), signed);
         }
 
+        private static void VerifyMembers(NamedTypeSymbol type)
+        {
+            var memberNames = type.MemberNames;
+            var allMembers = type.GetMembers();
```

Comment:
>allMembers [](start = 16, length = 10)

I think it would be good to assert that, each time we call GetMembers, we get the same instances for members.